src/main/java/net/sf/log4jdbc/StatementSpy.java

```diff
@@ -63,99 +81,88 @@ public Statement getRealStatement()
    * @param connectionSpy Connection that created this Statement.
    * @param realStatement real underlying Statement that this StatementSpy wraps.
    */
-  public StatementSpy(ConnectionSpy connectionSpy, Statement realStatement)
-  {
-    if (realStatement == null)
-    {
+  public StatementSpy(ConnectionSpy connectionSpy, Statement realStatement) {
+    if (realStatement == null) {
       throw new IllegalArgumentException("Must pass in a non null real Statement");
     }
-    if (connectionSpy == null)
-    {
+    if (connectionSpy == null) {
       throw new IllegalArgumentException("Must pass in a non null ConnectionSpy");
     }
     this.realStatement = realStatement;
     this.connectionSpy = connectionSpy;
 
     log = SpyLogFactory.getSpyLogDelegator();
 
-    if (realStatement instanceof CallableStatement)
-    {
+    if (realStatement instanceof CallableStatement) {
       reportReturn("new CallableStatement");
-    }
-    else if (realStatement instanceof PreparedStatement)
-    {
+    } else if (realStatement instanceof PreparedStatement) {
       reportReturn("new PreparedStatement");
-    }
-    else
-    {
+    } else {
       reportReturn("new Statement");
     }
   }
 
-  public String getClassType()
-  {
+  public String getClassType() {
     return "Statement";
   }
 
-  public Integer getConnectionNumber()
-  {
+  public Integer getConnectionNumber() {
     return connectionSpy.getConnectionNumber();
   }
 
   /**
    * Report an exception to be logged which includes timing data on a sql failure.
+   *
    * @param methodCall description of method call and arguments passed to it that generated the exception.
-   * @param exception exception that was generated
-   * @param sql SQL associated with the call.
-   * @param execTime amount of time that the jdbc driver was chugging on the SQL before it threw an exception.
+   * @param exception  exception that was generated
+   * @param sql        SQL associated with the call.
+   * @param execTime   amount of time that the jdbc driver was chugging on the SQL before it threw an exception.
    */
-  protected void reportException(String methodCall, SQLException exception, String sql, long execTime)
-  {
-    log.exceptionOccured(this, methodCall, exception, sql, execTime);
+  protected void reportException(String methodCall, SQLException exception, String sql, long execTime) {
+    // Throw, but do not log SQL exceptions if showing SQL arguments is disabled. Exception messages may contain
+    // sensitive data, so it's a caller's responsibility to handle exceptions properly.
+    if (showParams) log.exceptionOccured(this, methodCall, exception, sql, execTime);

```

Comment:
i think with this change we don't even see the exception also right?

Reply:
I really don't like the idea of completely omitting the log msg if it's reasonably easy instead to log a "safe" msg. Looking at the `exceptionOccured` method, this could be achieved by 1) passing the `showParams` param to `exceptionOccured` and 2) when false, update `exceptionOccured` to not include the exception or SQL in the logger call, and instead just append the class name of the exception to the msg, something like `header = header + ": exception occurred: " + e.getClass.getCanonicalName`

Reply:
@palanski Great that'd be preferable, @vivekragunathan any concerns?

Reply:
I am looking at the latest commit and now it is logging the exception type instead of the stack trace with message if the flag is off. Things are as usual if the flag is on. Am I reading it right @palanski ?

Reply:
The flag is `LOG4JDBC_SQL_SHOW_PARAMS`. I think we should come up with a better name taking into account it now has several functions. When it is `true`, it means "show arguments of prepared SQL statements" which can be generalised as "be unsafe". And when the flag is `false`, it means "be safe", and in this mode log4jdbc only log exception class names. So exception logging is in sync with SQL params logging as I understand it.

Reply:
Too late now to change the name but we can try that in a fresh year coming. But my question was "it is logging the exception type instead of the stack trace with message based on the flag value, right?"